rest_framework/relations.py

```diff
@@ -201,6 +201,23 @@ def get_url(self, obj, view_name, request, format):
             return None
 
         lookup_value = getattr(obj, self.lookup_field)
+
+        import numbers
+        from django.utils import six
+
+        if lookup_value is not None and not isinstance(lookup_value, six.string_types) \
+                                    and not isinstance(lookup_value, numbers.Number):
+
+            try:
+                lookup_value = getattr(lookup_value, self.lookup_url_kwarg)
+            except Exception:
+                lookup_value = getattr(lookup_value, 'pk')
+
+        if lookup_value is None or (not isinstance(lookup_value, six.string_types) \
+                                    and not isinstance(lookup_value, numbers.Number)) \
+                                    or '' == lookup_value:
+            return None
+
```

Comment:
The change previously made sense to me. I'm completely lost looking at this now. Why all the type checking, not at all clear to me if I was browsing this code what is going on or why.